vocabsieve/audio_player.py

```diff
@@ -0,0 +1,42 @@
+import os
+from threading import Thread
+from PyQt5.QtMultimedia import QMediaPlayer, QMediaContent
+from PyQt5.QtCore import QUrl
+from typing import Dict
+import requests
+from .forvo import *
+from .dictionary import *
+
+class AudioPlayer:
+  def __init__(self) -> None:
+    self.player = QMediaPlayer()
+
+  def crossplatform_play_sound(self, path):
+    self.player.setMedia(QMediaContent(path))
+    Thread(target=lambda: self.player.play()).start()
+
+  def play_audio(self, name: str, data: Dict[str, str], lang: str) -> str:
+    audiopath: str = data.get(name, "")
+    if not audiopath:
+        return ""
+
+    if not audiopath.startswith("https://"):
+        content = QUrl(audiopath)
+        self.crossplatform_play_sound(content)
```

Comment:
I wasn't sure how to test this if branch
 
I did test the same piece of code in a console and it worked for a local file

Reply:
Note that this is meant to be used with local files
which can be tested by using the local audio library feature, such as https://lingualibre.org/datasets/ (download a zip and unpack and import it in the local resource manager).
Note that they are allowed to be in any format your system can support. At a minimum it should include mp3, ogg, wav, and opus if your system have the codecs for it.

Reply:
I tested it, and this version does not work. What you want is:
`content = QUrl.fromLocalFile(audiopath)`

Reply:
Updated